Installing-and-Configuring/OS/Generic.md

```diff
@@ -0,0 +1,258 @@
+ provided by vendors.
+ 
+### Alternative Options
+
+Heads also supports booting directly from verified ISOs on a plain old
+ partition.  In this scenario heads will check the ISO against a corresponding
+ signature in the same location.  The ISO must be signed by a valid key in the Heads ROM or the boot process will fail.  You must verify the media immediately after download and sign it with your own key to establish a chain of trust.
+
+To use this feature, create a partition with a Heads compatible filesystem on a
+ USB storage device and copy the ISO images to it along with corresponding 
+ signatures.  The most common filesystems are: FAT and ext. The layout might
+ look like this:
+
+```shell
+/Qubes-R4.0-x86_64.iso
+/Qubes-R4.0-x86_64.iso.asc
+/Fedora-Workstation-Live-x86_64-27-1.6.iso
+/Fedora-Workstation-Live-x86_64-27-1.6.iso.sig
+/tails-amd64-3.7.iso
+/tails-amd64-3.7.iso.sig
+```
+
+Sign the ISOs with your own key:
+
+```shell
+gpg --output <iso_name>.sig --detach-sig <iso_name>
+```
+
+### Distro Specific ISO Boot
+
+Some distros require additional options for a successful boot directly from ISO.  See [Boot config files](#boot-config-files) for more information.
+1. Boot from USB by either running `usb-scan` or reboot into USB boot mode (hit
+ 'u' before the normal boot)
+1. Select the install boot option for your distro of choice and work through the
+ standard OS installation procedures (including setting up LUKS disk encryption
+ if desired)
+1. Reboot and your new boot options should be available to be chosen by
+ selecting 'm' at the boot screen
+
+`kexec_iso_add.txt` and `kexec_iso_remove.txt` are useful to inject the
+ appropriate kernel arguments to allow it to load properly. ISOs for Debian
+ require that `kexec_iso_add.txt` contains the following to load properly:
+
+```text
+findiso=${ISO_PATH}
+```
+
+Take a look at [https://mbusb.aguslr.com/howto.html](https://mbusb.aguslr.com/howto.html)
+ for more variations on the distro-specific ISO mounting command lines
+ requirements.  By default Heads uses two variants of this when booting from
+ ISO where a `kexec_iso_add.txt` is not specified:
+
+```text
+fromiso=/dev/disk/by-uuid/$DEV_UUID/$ISO_PATH iso-scan/filename=/${ISO_PATH}
+```
+
+
+Installation Choices
+====
+
+* partition scheme
+* partition encryption
+* bootloader
+
+partition scheme
+----
+
+Heads requires a separate /boot partition.  Other than that you may use any partition layout.
+
+
+partition encryption
+----
+
+The easiest approach is to leave the /boot partition unencrypted.  This is safe because heads will verify the files at boot.  Your other partitions may be encrypted or not at will (encryption is more secure) as the linux kernel booted by heads will be responsible for decrypting and mounting them.
+
+### Using TPM for decryption of drives
+
+You may seal your Disk Unlock Key using the TPM allowing you to use ensure only a boot passphrase and the proper PCR state can access the disk.
```

Comment:
WIP

>  to disk, while accepting to setup a Disk Unlock Key passphrase.

I dont understand the last part.  This is partly because I have never used the TPM with a disk passphrase--preferring to use USB token and /boot.  I understand the first steps to be:

The TPM will release a Disk Unlock Key by setting a default boot option and saving the changes to disk. 

but what is "while accepting to setup a Disk Unlock Key passphrase"?  Is that a menu in heads?

Reply:
> having different kernel modules will invalidaate the measurements.

Is this just the files in /boot or does heads measure any currently running modules?

Reply:
I pushed a new version of this section but github says my latest commits dont belong to any branch in this repo--which makes no sense to me.  Will let you know when I figure out what is going on.  https://github.com/osresearch/heads-wiki/pull/66/commits/9b569ffd2f3871db77aa306c5d60bb196425d56a

Reply:
This was also reworked and moved to OS/index where there was a duplicate section on TPM disk key.